src/LadderLine.java

```diff
@@ -0,0 +1,43 @@
+import java.util.Random;
+
+public class LadderLine {
+
+	private final StringBuilder ladder;
+	private static final Random ran = new Random();
+	private boolean previousStep = false;
+
+	public LadderLine(int userNumber) {
+		ladder = new StringBuilder();
+		int ladderRow = 2 * userNumber - 1;
+		makeLadder(ladderRow);
+	}
+
+	private void makeLadder(int ladderRow) {
+		for (int i = 1; i <= ladderRow; i++) {
+			makeLadderLine(i);
+		}
+	}
+
+	private void makeLadderLine(int rowIdx) {
+		if (rowIdx % 2 == 1) {
+			ladder.append("|");
+		} else {
+			boolean isStep = isStep();
+			previousStep = isStep;
+			ladder.append(isStep ? "-----" : "     ");
+		}
+	}
```

Comment:
```suggestion
	private void makeLadder(int ladderRow) {
		for (int rowIdx = 1; i <= ladderRow; i++) {
			makeLadderLine(rowIdx);
		}
	}

	private void makeLadderLine(int rowIdx) {
		if (rowIdx % 2 == 1) {
			ladder.append("|");
		} else {
			boolean isStep = isStep();
			previousStep = isStep;
			ladder.append(isStep ? "-----" : "     ");
		}
	}
```
위와 같이 for문에서 `i`를 쓰지 않고 `rowIdx` 로 쓰는 게 개인적으로 가독성이 좋아보이는데 `i`를 쓰는 것이 좋을까요 아니면 `rowIdx` 와 같이 쓰는 것이 좋을까요??

Reply:
요즘은 rowIdx를 더 추천합니다. 저는 소수파인 i파인데요 rowIdx가 더 좋아 보이는군요 ㅎㅎ